mivideo/mivideo_extract.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+import logging
+import os
+import time
+from typing import Union, Sequence, Dict
+
+import pandas as pd
+from google.cloud import bigquery
+from google.oauth2 import service_account
+from sqlalchemy.engine import ResultProxy
+
+import mivideo.queries as queries
+from db.db_creator import DBCreator
+from environ import ENV, CONFIG_DIR
+from run_jobs import ValidDataSourceName
+
+logger = logging.getLogger(__name__)
+
+SHAPE_ROWS: int = 0  # Index of row count in DataFrame.shape() array
+
+
+class MiVideoExtract(object):
+    udpKeyFilePath: str
+    credentials: service_account.Credentials
+    udpDb: bigquery.Client
+    appDb: DBCreator
+
+    def __init__(self):
+        self.udpKeyFilePath = os.path.join(CONFIG_DIR, ENV.get('mivideo', {}).get(
+            'service_account_json_filename'))
+        logger.debug(f'udpKeyFilePath: "{self.udpKeyFilePath}"')
+
+        self.credentials = service_account.Credentials.from_service_account_file(
+            self.udpKeyFilePath,
+            scopes=['https://www.googleapis.com/auth/cloud-platform'],
+        )
+
+        self.udpDb = bigquery.Client(
+            credentials=self.credentials,
+            project=self.credentials.project_id,
+        )
+
+        logger.debug(f'Connected to BigQuery project: "{self.udpDb.project}"')
+
+        DB_PARAMS = ENV['INVENTORY_DB']
+        APPEND_TABLE_NAMES = ENV.get('APPEND_TABLE_NAMES', ['job_run'])
```

Comment:
The default should include `data_source_status`. Note though that this will become unnecessary whenever issue #66 is resolved (hopefully soon).

Reply:
It'd be nice if `DBCreator` were a singleton that would be instantiated with `APPEND_TABLE_NAMES` only once in the running of this application.  That is, `run_jobs.py` would probably call it and all of the individual jobs' code could just use it.

Reply:
✅Fixed.

Reply:
Yeah, I think the `DBCreator` question could be raised in a separate issue. I could see each job entry function accepting a `DBCreator` object...

Reply:
Opened #99.